docs/networking.md

```diff
@@ -0,0 +1,100 @@
+* Include a `write_files` cloud-config snippet in the Elemental [MachineRegistration](machineregistration-reference) resource to create configuration files with content identified in the previous step
+
+:::info info
+The cloud-config configuration put in the Elemental [MachineRegistration](machineregistration-reference) is applied on the installed system only, not on the generated ISO/Image.
+This means that when booting from the Elemental ISO/Image the [MachineRegistration](machineregistration-reference) cloud-config is not applied on the host: it will be applied only after the OS is installed and rebooted.
+
+Host configuration during Elemental ISO/Image boot is possible adding the cloud-config customization in the [SeedImage](seedimage-reference) resource instead of the [MachineRegistration](machineregistration-reference) one.
+:::
+
+### Identify NetworkManager connection profile configuration files content
+NetworkManager supports multiple connection profile storage formats.
+While one could focus on any of the supported configuration file plugins, the [keyfile plugin](https://networkmanager.dev/docs/api/latest/nm-settings-keyfile.html) is the one we recommend: it allows to store all the possible connection profile data and has a quite simple syntax.
+
+NetworkManager keyfiles can be written directly following the [nm-settings-keyfile man page](https://networkmanager.dev/docs/api/latest/nm-settings-keyfile.html).
+
+Anyway, it could be easier to instruct NetworkManager to configure a connection profile if a live system is available: NetworkManager then creates the keyfiles for us in the `/etc/NetworkManager/system-connections/` directory.
+
+There are multiple different ways to interact with NetworkManager and create connection profiles. The most used and handy configuration tools include:
+* [nmcli](https://networkmanager.dev/docs/api/latest/nmcli.html) - the NetworkManager CLI
+* [nmtui](https://networkmanager.dev/docs/api/latest/nmtui.html) - the NetworManager Text User Interface
+* the UI applets present in all the major linux desktops environments
+
+Finally, to generate NetworkManager keyfiles _offline_ using a declarative approach, one could use the [nm-configurator](https://github.com/suse-edge/nm-configurator) project.
+
+```shell title="Example: generate a static IPv4 ethernet connection porfile with nmcli" showLineNumbers
+nmcli connection add \
+  con-name fixed-ip\
+  type ethernet \
+  ipv4.method manual \
+  ipv4.addresses 192.168.1.2/24 \
+  ipv4.gateway 192.168.1.1 \
+  ipv4.dns 192.168.1.1
+```
+After running the above command, you will find your keyfile at
+`/etc/NetworkManager/system-connections/fixed-ip.nmconnection`.
+
+### Elemental cloud-config
+The NetworkManager connection profile keyfiles can be injected it in a [MachineRegistration](machineregistration-reference) using the `write_files` module.
+
+The `content`` can be either plain text or base64 encoded (`encoding: b64`).
```

Comment:
Wrong formatting I guess, I checked the preview and it looks bad:
![image](https://github.com/rancher/elemental-docs/assets/6025636/2e149233-c433-46af-af2c-ae7344d55ff9)

Reply:
good catch! 
```suggestion
The `content` can be either plain text or base64 encoded (`encoding: b64`).
```